src/synthesis.cpp

```diff
@@ -235,8 +251,21 @@ static int GetPulseLocationsForTimeBase(const double *interpolated_f0,
   for (int i = 0; i < y_length - 1; ++i) {
     if (wrap_phase_abs[i] > world::kPi) {
       pulse_locations[number_of_pulses] = time_axis[i];
-      pulse_locations_index[number_of_pulses] = static_cast<int>
-        (matlab_round(pulse_locations[number_of_pulses] * fs));
+      pulse_locations_index[number_of_pulses] = i;
+
```

Comment:
This change assumes `pulse_locations[number_of_pulses] = time_axis[i] = i / fs`, therefore `pulse_location_index[number_of_pulses] = int(round(pulse_locations[number_of_pulses] * fs)) = i`.

This change is not strictly needed, but just a simplification that makes understanding the code easier. In particular, the use of `round()` may seem like `pulse_locations_index` is the _rounded_ integer pulse location, but in reality it is the _floored_ integer pulse location. Also, in the code below `wrap_phase[i]` and `wrap_phase[i + 1]` are used, implicitly assuming the pulse location is `i`.

Please correct me if my assumptions are wrong, I may have misunderstood some things.